trl/trainer/grpo_trainer.py

```diff
@@ -839,22 +846,34 @@ def _generate_and_score_completions(
         self._metrics[mode]["reward_std"].append(std_grouped_rewards.mean().item())
 
         if (
-            self.log_completions
-            and self.state.global_step % self.args.logging_steps == 0
-            and "wandb" in self.args.report_to
+            self.log_completions_steps is not None
+            and self.log_completions_steps > 0
+            and self.state.global_step % self.log_completions_steps == 0
         ):
-            import pandas as pd
-
-            # For logging
-            table = {
-                "step": [str(self.state.global_step)] * len(rewards),
-                "prompt": gather_object(prompts_text),
-                "completion": gather_object(completions_text),
-                "reward": rewards.tolist(),
-            }
-            df = pd.DataFrame(table)
-
-            if wandb.run is not None and self.accelerator.is_main_process:
+            prompts_to_log: list[str] = gather_object(prompts_text)
+            completions_to_log: list[str] = gather_object(completions_text)
+            pairs_to_log = random.sample(list(zip(prompts_to_log, completions_to_log)), 5)
+            print_prompt_completions_sample(
+                [prompt for prompt, _ in pairs_to_log],
+                [completion for _, completion in pairs_to_log],
+                self.state.global_step,
+            )
+            if (
+                self.args.report_to
+                and "wandb" in self.args.report_to
+                and wandb.run is not None
+                and self.accelerator.is_main_process
```

Comment:
I think this condition should be moved up

Reply:
Up where? 😅

Reply:
```python
if self.accelerator.is_main_process:
    print_prompt_completions_sample(
        [prompt for prompt, _ in pairs_to_log],
        [completion for _, completion in pairs_to_log],
        self.state.global_step,
    )
    if (
    self.args.report_to
    and "wandb" in self.args.report_to
    and wandb.run is not None
    ):
        ...

Reply:
Oh true, I missed that. Thanks.